File: api/src/Domain/ProductCatalog/PadSeries/Services/PadSeriesService.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using DetailingArsenal.Domain.Users;

namespace DetailingArsenal.Domain.ProductCatalog {
    public interface IPadSeriesService : IService {
        Task<PadSeries> GetById(Guid id);
        Task<PadSeries> Create(PadSeriesCreate create, User user);
        Task<PadSeries> Update(PadSeries series, PadSeriesUpdate update, User user);
        Task Delete(PadSeries series, User user);
    }

    public class PadSeriesService : IPadSeriesService {
        IPadSeriesRepo repo;

        public PadSeriesService(IPadSeriesRepo repo) {
            this.repo = repo;
        }

        public async Task<PadSeries> GetById(Guid id) {
            var series = await repo.FindById(id);
            return series ?? throw new EntityNotFoundException();
        }

        public async Task<PadSeries> Create(PadSeriesCreate create, User user) {
            var pads = create.Pads.Select(p => new Pad(p.Category, p.Name, p.Image)).ToList();

            var series = new PadSeries(
                create.Name,
                create.BrandId,
                pads
            );

            await repo.Add(series);
            return series;
        }

        public async Task<PadSeries> Update(PadSeries series, PadSeriesUpdate update, User user) {
            series.Name = update.Name;
            series.BrandId = update.BrandId;
            series.Pads = update.Pads.Select(p => new Pad(p.Category, p.Name, p.Image)).ToList();

            await repo.Update(series);
            return series;
        }

        public async Task Delete(PadSeries series, User user) {
            await repo.Delete(series);
        }
    }
}

File: app/src/api/product-catalog/pad-series/services/pad-series-service.ts
import { Pad } from '@/api/product-catalog/pad-series/data-transfer-objects/pad';
import { PadSeriesCreateRequest } from '@/api/product-catalog/pad-series/data-transfer-objects/requests/pad-series-create-request';
import { PadSeriesUpdateRequest } from '@/api/product-catalog/pad-series/data-transfer-objects/requests/pad-series-update-request';
import { PadSeries, Brand } from '@/api';
import { http } from '@/api/core/http';

export class PadSeriesService {
    async get(): Promise<PadSeries[]> {
        const res = await http.get('product-catalog/pad');
        return (res.data as any[]).map(d => this._map(d));
    }

    async create(create: PadSeriesCreateRequest) {
        const res = await http.post('product-catalog/pad', create);
        const ps = this._map(res.data);
        return ps;
    }

    async update(update: PadSeriesUpdateRequest) {
        const res = await http.put(`product-catalog/pad/${update.id}`, update);
        const ps = this._map(res.data);
        return ps;
    }

    async delete(id: string) {
        await http.delete(`product-catalog/pad/${id}`);
    }

    _map(ps: any): PadSeries {
        const series = new PadSeries(ps.id, ps.name, new Brand(ps.brand.id, ps.brand.name));

        // series.pads = (ps.pads as any[]).map(p => new Pad(p.id, p.category, series, p.name, p.image));

        return series;
    }
}
